curriculum/challenges/english/25-front-end-development/quiz-javascript-fundamentals/66edcd875b0d91de1fbbb492.md

```diff
@@ -17,439 +17,532 @@ To pass the quiz, you must correctly answer at least 18 of the 20 questions belo
+`handlePermission`
 
 ---
 
-To create documentation for code.
+`permission`
 
 #### --answer--
 
-To detect potential errors and style issues in code.
+`hasPermission`
 
 ### --question--
 
 #### --text--
 
-Which of the following is NOT a linter for JavaScript?
+Which of the following is best name for a function that runs when a user submits a form?
 
 #### --distractors--
 
-ESLint
+`hasSubmitted()`
 
 ---
 
-standardJS
+`formSubmit()`
 
 ---
 
-quick-lint-js
+`didSubmit()`
 
 #### --answer--
 
-Babel
+`handleSubmit()`
 
 ### --question--
 
 #### --text--
 
-Which of these describes accessibility?
+What is a sparse array?
 
 #### --distractors--
 
-Making sure code is accessible to other developers.
+An array where one or more indices have a value of `null`.
 
 ---
 
-Writing secure code.
+An array where one or more indices have a value of `undefined`.
 
 ---
 
-Controlling who can access certain data.
+An array where one or more indices have a value of an empty object literal.
 
 #### --answer--
 
-Using events wisely to avoid locking out specific control types (keyboard, touchscreen users)
+An array where one or more indices are not assigned a value.
 
 ### --question--
 
 #### --text--
 
-Why is using the `var` keyword considered bad practice in modern JavaScript?
+In which of the following examples will `arr` NOT be a sparse array?
 
 #### --distractors--
 
-It's slower than `let` and `const`
+```js
+const arr = new Array(5)
+```
 
 ---
 
-It causes a syntax error.
+```js
+const arr = [1, 2, 3];
+arr[4] = 4
+```
 
 ---
 
-It doesn't allow type checking.
+```js
+const arr = [1, 2, 3, , 5]
+```
 
 #### --answer--
 
-It has function scope instead of block scope, which can lead to unexpected behaviors.
+```js
+const arr = Array.from({ length: 5 });
+```
 
 ### --question--
 
 #### --text--
 
-In JavaScript, how is memory for variables typically allocated?
+What is the primary purpose of a linter?
 
 #### --distractors--
 
-Manually by the developer.
+To format code automatically.
 
 ---
 
-By the browser.
+To optimize code for performance.
 
 ---
 
-By the server.
+To create documentation for code.
 
 #### --answer--
 
-Automatically by the runtime engine.
+To detect potential errors and issues in code.
 
 ### --question--
 
 #### --text--
 
-What is hoisting in JavaScript?
+What is a benefit of using a formatter in a codebase?
 
 #### --distractors--
 
-Raising errors to the top of the code.
+They allow developers to compile code more efficiently.
 
 ---
 
-The process of creating new variables.
+They convert code to a specific programming language.
 
 ---
 
-Automatically moving `var` declarations to the top of the block.
+They detect logical errors.
 
 #### --answer--
 
-The process of moving variable declarations to the top of their scope.
+They ensure a consistent code style.
 
 ### --question--
 
 #### --text--
 
-What is the purpose of modules in JavaScript?
+In JavaScript, how is memory for variables typically allocated?
```

Comment:
here you ask "how" but the answers are to "by whom"

Reply:
Changed it up:

```
--text--

In JavaScript, who or what is typically responsible for allocating memory to variables?

--distractors--

The developer.

The operating system.

The server.

--answer--

The runtime engine.
```